Sources/TuistPlugin/Service/PluginService.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// A protocol defining a service for interacting with plugins.
+public protocol PluginServicing {
+    /// Loads the plugins and returns them.
+    /// - Throws: An error if couldn't load a plugin.
+    /// - Returns: The loaded `Plugins` representation.
+    func loadPlugins() throws -> Plugins
+}
+
+/// A default implementation of `PluginServicing`.
+public class PluginService: PluginServicing {
+    private let config: Config
+    private let graphLoader: GraphLoading
+    private let fileHandler: FileHandling
+    private let gitHandler: GitHandling
+
+    /// Creates a `PluginService`.
+    /// - Parameters:
+    ///   - config: The loaded `Config` used to parse plugins.
+    ///   - graphLoader: A graph loader for loading plugins.
+    ///   - fileHandler: A file handler for creating plugin directories/related files.
+    ///   - gitHandler: A git handler for cloning and interacting with remote plugins.
+    public init(
+        config: Config,
+        graphLoader: GraphLoading,
+        fileHandler: FileHandling = FileHandler.shared,
+        gitHandler: GitHandling = GitHandler()
+    ) {
+        self.config = config
+        self.graphLoader = graphLoader
+        self.fileHandler = fileHandler
+        self.gitHandler = gitHandler
+    }
+
+    public func loadPlugins() throws -> Plugins {
+        let pluginPaths = try fetchPlugins()
+        let pluginManifests = try pluginPaths.map(graphLoader.loadPlugin)
+
+        var projectDescriptionHelpers = [CustomProjectDescriptionHelpers]()
+
+        zip(pluginManifests, pluginPaths).forEach { plugin, path in
```

Comment:
I think this can be further simplified into something like this:
```swift
let projectDescriptionHelpers = zip(pluginManifests, pluginPaths)
.map { ... return .init(name: name, path: path) }
```

`projectDescriptionHelpers` will then no longer have to mutable 🙂